grafana/dashboards/timeline.json

```diff
@@ -0,0 +1,718 @@
ence_id IS NULL THEN CONCAT('new?lat=', TP2.latitude, '&lng=', TP2.longitude)\r\n\t\tWHEN T2.start_geofence_id IS NOT NULL THEN CONCAT(T2.start_geofence_id, '/edit')\r\n\tEND AS end_path,\r\n\tCOALESCE(start_geofence.name, CONCAT_WS(', ', COALESCE(start_address.name, nullif(CONCAT_WS(' ', start_address.road, start_address.house_number), '')), start_address.city)) AS \"Start Address\",\r\n\tCOALESCE(end_geofence.name, CONCAT_WS(', ', COALESCE(end_address.name, nullif(CONCAT_WS(' ', end_address.road, end_address.house_number), '')), end_address.city)) AS \"End Address\",\r\n\tconvert_km(TP1.odometer::INTEGER, '$length_unit') AS odometer_$length_unit,\r\n\tconvert_km((TP2.odometer - TP1.odometer)::INTEGER, '$length_unit') AS distance_$length_unit,\r\n\tconvert_km(((TP2.ideal_battery_range_km + (TP2.odometer - TP1.odometer)) - TP1.ideal_battery_range_km)::INTEGER, '$length_unit')::INTEGER * efficiency AS \"kWh\",\r\n\tconvert_km(((TP2.ideal_battery_range_km + (TP2.odometer - TP1.odometer)) - TP1.ideal_battery_range_km)::INTEGER, '$length_unit') AS range_diff_$length_unit,\r\n\tTP2.battery_level AS \"SoC\",\r\n\tTP2.battery_level-TP1.battery_level AS \"SoC Diff\",\r\n\tT1.outside_temp_avg AS outside_temp_avg_$temp_unit\r\nFROM\r\n\tdrives AS T1\r\n\tLEFT OUTER JOIN drives AS T2 on T1.id + 1 = T2.id\r\n\tLEFT OUTER JOIN positions AS TP1 on T1.end_position_id = TP1.id\r\n\tLEFT OUTER JOIN positions AS TP2 on T2.start_position_id = TP2.id\r\n\tLEFT JOIN addresses start_address ON T1.end_address_id = start_address.id\r\n\tLEFT JOIN addresses end_address ON T2.start_address_id = end_address.id\r\n\tLEFT JOIN geofences start_geofence ON T1.end_geofence_id = start_geofence.id\r\n\tLEFT JOIN geofences end_geofence ON T2.start_geofence_id = end_geofence.id\r\n\tJOIN cars ON cars.id = T2.car_id\r\nWHERE\r\n\t$__timeFilter(T1.end_date)\r\n\tAND T1.car_id=$car_id \r\n\tAND T2.car_id=$car_id \r\n\tAND TP2.car_id=$car_id \r\n\tAND TP1.car_id=$car_id \r\n\tAND TP2.odometer - TP1.odometer > 0.5\r\n\tAND T1.end_address_id <> T2.start_address_id AND T1.end_geofence_id <> T2.start_geofence_id\r\n\tAND '❓ Missing' in ($action_filter)\r\n\tAND ('Start Address' like '%$text_filter%' or 'End Address' like '%$text_filter%')\r\n\r\nUNION\r\nSELECT\r\n\tstart_date AS \"Start\",\r\n\tstart_date AS \"End\",\r\n\tROUND(EXTRACT(EPOCH FROM start_date))*1000 AS start_date_ts, \r\n\tROUND(EXTRACT(EPOCH FROM start_date))*1000 AS end_date_ts, \r\n\t'💾 Update' AS \"Action\",\r\n\tnull AS distance_$length_unit,\r\n\tnull AS start_path,\r\n\tnull AS end_path,\r\n\tversion AS \"Start Address\",\r\n\t'' AS \"End Address\",\r\n\tnull AS odometer_$length_unit,\r\n\tnull AS \"Distance\", \r\n\tnull AS \"kWh\",\r\n\tnull AS range_diff_$length_unit,\r\n\tnull AS \"SoC\",\r\n\tnull AS \"SoC Diff\",\r\n\tnull AS outside_temp_avg_$temp_unit\r\nFROM updates\r\nWHERE \r\n\t$__timeFilter(start_date)\r\n\tAND car_id = $car_id \r\n\tAND '💾 Update' in ($action_filter)\r\n\r\nORDER BY \"Start\" DESC\r\n",
```

Comment:
Nitpick: The actions are named somewhat inconsistently. How about this:

"🔋 charges" -> "🔋 Charging"
"🚗 Drive" -> "🚗 Driving"
"💾 Update" -> "💾 Updating"

Reply:
OK